### ISSUE TITLE
Spike - Are there a11y advantages to React-Router?

### ISSUE THREAD

**Issue by tristan-orourke** [GCTC-NTGC/gc-digital-talent#2368]
# Description
We're doing frontend routing with a custom hook and [UniversalRouter](https://github.com/kriasoft/universal-router). Are there any major features, especially around accessibility, that [React Router](https://reactrouter.com/) offers that we are currently missing?

Timeboxed to 1-2 hours.

**Comment by tristan-orourke** [comment]
Just from a cursory search, I wonder if it has to do with this - announcing page navigations for screen readers? https://estevanmaito.me/blog/accessible-navigation-using-react-router

**Comment by patcon** [comment]
What I recall of reach-router and react-router differences and fork/merge:

- https://reach.tech/router/accessibility
- https://reach.tech/router/credits (react vs reach)
  - polyfills aren't supported (re: small size) re: #2361 
  - accessibility
- https://reactrouter.com/docs/en/v6/upgrading/reach
  - react-router v6 incorporates many APIs of reach-router
  - accessibility: "Stop doing not-good-enough focus management by default."

Both reach and react-router now handle focus management in much more accessible ways, for both client-side and server-size.

Open questions:
1. Does accessibility concerns for "old" router versions only apply to server-side (of which we don't currently rely)? Will we ever use server-side rendering?
2. How will out browser support goals intersect with router choice?

**Comment by patcon** [comment]
Still need to read this, but mentions (circa 2019) that **server-side rendering is highly recommended** for accessibility in govuk react sites: https://github.com/govuk-react/govuk-react/blob/main/ACCESSIBILITY.md

**Comment by patcon** [comment]
Via [React Router v6 in Three Minutes](https://betterprogramming.pub/react-router-v6-in-two-minutes-a7a2963e2340) (Feb 2020):

https://twitter.com/mjackson/status/1229156979714605056

> what was the biggest factor that led to this change [in bundle size down 70%]?

> A couple of things:
> - dropping support for anything older than IE11
> - dropping support for React < 16.8
> - using Google Closure Compiler
> - writing better code 😅

**Comment by patcon** [comment]
Not specifically a11y, but here's another downside of not having react-router.

I'm finding it hard to imagine how component state is exercised for stories whose components contain logic based on `location.pathname`. As it stands, we'd need to make pathname or location optionally injectable into each component using that logic. This is due to how we're implementing universal-router without any context to override:

https://github.com/GCTC-NTGC/gc-digital-talent/blob/156e5e3fbe4f9245bc2006ae70abec74004310ec/frontend/common/src/helpers/router.tsx#L8-L22

This is what using `<MemoryRouter>` in a story looks like:
https://chestozo.medium.com/how-to-mock-location-inside-storybook-stories-76a7c0705354

Since our `useLocation()` hook hardcodes the use of `createBrowserHistory()`, we can't easily use [`createMemoryHistory()`](https://github.com/remix-run/history/blob/dev/docs/api-reference.md#creatememoryhistory) like react-router does. I believe this is what is offered by the extra layer of `<Router>`, which adds context that stores this kind of stuff, and can be modified.

**Comment by tristan-orourke** [comment]
The answer to the spike seems to be YES
